### PR TITLE
Adding evaluation script for automatic pipeline

### PR DESCRIPTION
**Pull request by shanybar**
The new script is eval_auto_pipeline.py.
There are usage instructions in the top of the script.
The script evaluate entity and predicate mentions extraction, and also entity and predicate coreference.
I added the corresponding input files of the dev/test (see in data/baseline).
I also added the input files and the gold files of the big stories (see in date/big_stories).
I added Hitesh's changes to the script eval_predicate_coref.py, but I didn't integrate it yet with the automatic pipeline.

### CODE REVIEW

**Comment by gabrielStanovsky** [comment]
* I think it would be good to add a README.md file to each of the new folders describing what's in them (e.g., "raw sentences for the development set")
* I'm not sure, but I think we use the term "event" instead of "story" elsewhere in the repo?

**Comment by gabrielStanovsky** [comment]
In addition, can you please add a .md file with the output of the evaluation script? That way we can keep track of the performance changes.

File: src/agreement/predicate_mention.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from filter_propositions import filter_verbal, filter_non_verbal
 
 
-def compute_predicate_mention_agreement(graph1, graph2):
+def compute_predicate_mention_agreement(graph1, graph2,ignore_filtering=False):
```
Comment (gabrielStanovsky) [inline src/agreement/predicate_mention.py]: Can you add documentation about the new ```ignore_filtering``` argument?
Does it have to have a default value, I generally prefer to do without these, if possible.

File: src/baseline_automatic_pipeline_system/eval_auto_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+from parse_okr_info import get_raw_sentences_from_file,auto_pipeline_okr_info
+from eval_entity_coref import eval_entity_coref_between_two_graphs
+from eval_predicate_coref import eval_predicate_coref_between_two_graphs
+from eval_predicate_mention import evaluate_predicate_mention_between_two_graphs
+
+def eval_auto_pipeline(input_folder,gold_folder):
+    """
+    Evaluate entity and predicate coreference for auto-created graphs and their gold graphs
+    :param input_folder: Path to the input files folder (text files)
+    :param gold_foldar: Path to the gold graphs folder (xml files)
```
Comment (gabrielStanovsky) [inline src/baseline_automatic_pipeline_system/eval_auto_pipeline.py]: Nit: Typo in ```gold_foldar```

File: src/baseline_automatic_pipeline_system/eval_auto_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    predicate_coref_scores = []
+    entity_coref_scores = []
+    predicate_mention_scores = []
+    entity_mention_scores = []
+
+
+    for input_file,gold_file in zip(sorted_input,sorted_gold):
+        print 'Evaluate input file '+ input_file + ' with gold file ' + gold_file
+        sentences = get_raw_sentences_from_file(input_folder + '/' + input_file)
```
Comment (gabrielStanovsky) [inline src/baseline_automatic_pipeline_system/eval_auto_pipeline.py]: Better to use:
```python
os.path.join(input_folder, input_file)
```

Comment (gabrielStanovsky) [reply src/baseline_automatic_pipeline_system/eval_auto_pipeline.py]: Also in other places where you concatenate a file to a folder

File: src/baseline_system/clustering_common.py [hunk 4]
```diff
@@ -25,3 +25,34 @@ def cluster_mentions(mention_list, score):
...
+    clusters = []
+
+    for mention in mention_list:
+        found_cluster = False
+        max_score = 0
+        max_cluster = None
+        for cluster in clusters:
+            cluster_score = score(mention, cluster)
+            if cluster_score > 0.2:
+                # cluster.add(mention)
```
Comment (gabrielStanovsky) [inline src/baseline_system/clustering_common.py]: Better to remove commented out code.

File: src/baseline_system/clustering_common.py [hunk 5]
```diff
@@ -25,3 +25,34 @@ def cluster_mentions(mention_list, score):
             clusters.append(set([mention]))
 
     return clusters
+
+def cluster_mentions_with_max_cluster(mention_list, score):
+    """
```
Comment (gabrielStanovsky) [inline src/baseline_system/clustering_common.py:30]: Where is this function coming from? Hitesh?

File: src/baseline_system/clustering_common.py [hunk 6]
```diff
@@ -25,3 +25,34 @@ def cluster_mentions(mention_list, score):
...
+        max_cluster = None
+        for cluster in clusters:
+            cluster_score = score(mention, cluster)
+            if cluster_score > 0.2:
+                # cluster.add(mention)
+                if cluster_score > max_score:
+                    max_score = cluster_score
+                    max_cluster = cluster
+                    found_cluster = True
+                #break
```
Comment (gabrielStanovsky) [inline src/baseline_system/clustering_common.py]: Better to remove commented out code.

File: src/baseline_system/eval_entity_coref.py [hunk 7]
```diff
@@ -48,6 +50,7 @@ def evaluate_entity_coref(test_graphs):
         # Cluster the entities
         entities = [(str(mention), unicode(mention.terms)) for entity in graph.entities.values() for mention in
                     entity.mentions.values()]
+        # clusters = cluster_mentions_with_max_cluster(entities, score)
```
Comment (gabrielStanovsky) [inline src/baseline_system/eval_entity_coref.py]: Better to remove commented out code.

File: src/baseline_system/eval_predicate_coref.py [hunk 8]
```diff
@@ -51,6 +66,126 @@ def evaluate_predicate_coref(test_graphs):
...
+#     For similarity we compare any two non-stop words of the two mentions for same_synset, partial WordNet match, fuzzy string similarity.
+#     :param prop: the mention
+#     :param cluster: the cluster
+#     :return: a numeric value denoting the similarity between the mention to the cluster
+#     """
+#
+#     # if you want to compare the dependency heads only for similarity, use this:
+#     #return len([other for other in cluster if similar_words(other[1],prop[1])]) / (1.0 * len(cluster))
+#
+#     return len([other for other in cluster if some_word_match(other[2], prop[2])]) / (1.0 * len(cluster))
```
Comment (gabrielStanovsky) [inline src/baseline_system/eval_predicate_coref.py]: Better to remove commented out code (or functions :))

File: src/baseline_system/eval_predicate_mention.py [hunk 9]
```diff
@@ -23,6 +23,14 @@
 
 logging.basicConfig(level = logging.INFO)
 
+def evaluate_predicate_mention_between_two_graphs(pred_graph,gold_graph):
```
Comment (gabrielStanovsky) [inline src/baseline_system/eval_predicate_mention.py]: Documentation missing

**Review by gabrielStanovsky — COMMENTED** [review]
Looks good to me, other than what I pointed below. 
@kleinay , can you give the changes in [parse_okr_info](src/baseline_automatic_pipeline_system/parse_okr_info.py) a look, as this is your code?
I specifically think it would be nice to maintain the current evaluation numbers in some file.

File: src/baseline_system/clustering_common.py (hunk 5 above)
Comment (shanybar) [reply src/baseline_system/clustering_common.py:30]: No, I created it in order to examine some changes in clustering algorithm.

**Review by kleinay — COMMENTED** [review]
I reviewed the changes at parse_okr_info.py, they all look fine to me.

**Comment by gabrielStanovsky** [comment]
Thanks for adding the requested changes. 
Merging now.

**Pull request merged**